AdblockPlusSafari/AppDelegate.m

```diff
@@ -149,12 +145,20 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
 
     AdblockPlusExtras *adblockPlus = object;
     if (!adblockPlus.updating) {
-
       for (NSDictionary *backgroundFetch in self.backgroundFetches) {
 
-        // The date of the last known successful update
-        NSDate *lastUpdate = backgroundFetch[@"lastUpdate"];
-        BOOL updated = !!lastUpdate && [adblockPlus.lastUpdate compare:lastUpdate] == NSOrderedDescending;
+        BOOL updated = NO;
+
+        NSDictionary<NSString *, id> *filterLists = backgroundFetch[@"filterLists"];
+        for (NSString *filterListName in filterLists) {
+          NSDictionary<NSString *, id> *filterList = filterLists[filterListName];
+          // The date of the last known successful update
+          NSDate *lastUpdate = filterList[@"lastUpdate"];
+
+          NSDate *currentLastUpdate = self.adblockPlus.filterLists[filterListName][@"lastUpdate"];
+
+          updated = updated || (!lastUpdate && !!currentLastUpdate) || (!!currentLastUpdate && [currentLastUpdate compare:lastUpdate] == NSOrderedDescending);
```

Comment:
Nit: Is the `!!` necessary here? The `&&` operator should have that effect already.